settings.py

```diff
@@ -131,6 +131,7 @@
 CSRF_SOFT_MODE = True
 
 MIDDLEWARE_CLASSES = [
+    'corehq.middleware.NoCacheMiddleware',
     'django.middleware.common.CommonMiddleware',
     'django.contrib.sessions.middleware.SessionMiddleware',
```

Comment:
Looks good, but are you sure that NoCacheMiddleware should come before SessionMiddleware and not after it? Just wondering whether you're removing Set-Cookie only for SessionMiddleware to stick it back in.